libs/dh/wholesale/feature-settlement-reports/src/lib/tabs/dh-wholesale-settlements-reports-tabs-balance.component.ts

```diff
@@ -164,10 +165,14 @@ export class DhWholesaleSettlementsReportsTabsBalanceComponent implements OnInit
 
     this.subscriptionGridAreasForFilter = this.gridAreasForFilterQuery.valueChanges.subscribe({
       next: (result) => {
-        this.gridAreas = (result.data?.gridAreas ?? []).map((g) => ({
+        const gridAreasClone = structuredClone(result.data?.gridAreas ?? []);
```

Comment:
+1 for structuredClone, instead of JSON.stringfy

Reply:
Learned about it recently. Really cool that we have such function for deep cloning supported in all browsers.